tests/components/deconz/test_init.py

```diff
@@ -97,6 +97,12 @@ async def test_setup_entry_multiple_gateways(
     assert hass.data[DECONZ_DOMAIN][config_entry.entry_id].master
     assert not hass.data[DECONZ_DOMAIN][config_entry2.entry_id].master
 
+    await config_entry.async_unload(hass)
+    await hass.async_block_till_done()
```

Comment:
See https://github.com/home-assistant/core/actions/runs/4685768090/jobs/8303233507
I have a race condition in #90850, where the second config entry is still being unloaded whilst also getting updated from the first config entry unload.

The race condition can be replicated with:
```python
    await asyncio.gather(
        config_entry.async_unload(hass), config_entry2.async_unload(hass)
    )
```

Reply:
I have opened an alternative in #91327